verification.py

```diff
@@ -878,6 +936,192 @@ def test_sweep_tau_ex3(self):
+            model=[None, (tau, sweep), (None, "hudson")],
+            recombination_rate=recombination_rate,
+            sequence_length=seq_length,
+            num_replicates=num_replicates,
+        )
+        wins = range(0, int(seq_length + 1), int(seq_length / 20))
+        mids = np.zeros(len(wins) - 1)
+        for i in range(len(wins) - 1):
+            mids[i] = (wins[i + 1] + wins[i]) / 2
+        msp_win_pis = []
+        slim_win_pis = []
+        data = collections.defaultdict(list)
+        for ts in replicates:
+            t_mrca = np.zeros(ts.num_trees)
+            for tree in ts.trees():
+                t_mrca[tree.index] = tree.time(tree.root)
+            data["tmrca_mean"].append(np.mean(t_mrca))
+            data["num_trees"].append(ts.num_trees)
+            mutated_ts = msprime.sim_mutations(ts, rate=1e-8)
+            data["pi"].append(mutated_ts.diversity().reshape((1,))[0])
+            data["model"].append("msp")
+            msp_num_samples = ts.num_samples
+            msp_win_pis.append(mutated_ts.diversity(windows=wins))
+        slim_script = self.output_dir / "slim_script.txt"
+        outfile = self.output_dir / "slim.trees"
+        slim_args["OUTFILE"] = str(outfile)
+        write_sweep_slim_script(slim_script, slim_args)
+
+        cmd = _slim_executable + [slim_script]
+        for _ in range(kwargs["num_replicates"]):
+            subprocess.check_output(cmd)
+            ts = pyslim.load(outfile)
+            rts = sample_recap_simplify(
+                ts, sample_size, pop_size, recombination_rate, 1e-8
+            )
+            assert rts.num_samples == msp_num_samples
+
+            t_mrca = np.zeros(rts.num_trees)
+            for tree in rts.trees():
+                t_mrca[tree.index] = tree.time(tree.root)
+
+            data["tmrca_mean"].append(np.mean(t_mrca))
+            data["num_trees"].append(rts.num_trees)
+            slim_win_pis.append(rts.diversity(windows=wins))
+            data["pi"].append(rts.diversity().reshape((1,))[0])
+            data["model"].append("slim")
+        df = df.append(pd.DataFrame(data))
+
+        df_slim = df[df.model == "slim"]
+        df_msp = df[df.model == "msp"]
+        for stat in ["tmrca_mean", "num_trees", "pi"]:
+            v1 = df_slim[stat]
+            v2 = df_msp[stat]
+            sm.graphics.qqplot(v1)
+            sm.qqplot_2samples(v1, v2, line="45")
+            pyplot.xlabel("msp")
+            pyplot.ylabel("SLiM")
+            f = self.output_dir / f"{stat}.png"
+            pyplot.savefig(f, dpi=72)
+            pyplot.close("all")
+            plot_stat_hist(v1, v2, v1.name, v2.name)
+            f = self.output_dir / f"{stat}.hist.png"
+            pyplot.savefig(f, dpi=72)
+            pyplot.close("all")
+        pyplot.plot(mids, np.array(msp_win_pis).mean(axis=0), label="msp")
+        pyplot.plot(mids, np.array(slim_win_pis).mean(axis=0), label="slim")
+        pyplot.title(f"tau: {tau}")
```

Comment:
Another nitpicking point: I would suggest adding the xlabel and ylabel to this plot